haskell/src/Ligo/BaseDAO/Types.hs

```diff
@@ -625,6 +636,13 @@ mkFullStorage admin vp qt extra mdt now tokenAddress cEps = FullStorage
   , fsConfig  = mkConfig (argDef #customEps [] cEps)
   }
 
+setExtra :: forall a. NicePackedValue a => MText -> a -> FullStorage -> FullStorage
+setExtra key v (s@FullStorage {..}) = let
+  (BigMap oldExtra) = unDynamicBigMap $ sExtra fsStorage
+  newExtra = BigMap $ M.insert key (lPackValueRaw v) oldExtra
+  newStorage = fsStorage { sExtra = DynamicRecBigMap newExtra }
+  in s { fsStorage = newStorage }
```

Comment:
Nitpick: the `let` indentation is a bit odd, it should be on the line below and the other lines moved to the right.

Otherwise I suggest using `where` instead:
```suggestion
setExtra key v (s@FullStorage {..}) = s { fsStorage = newStorage }
  where
    (BigMap oldExtra) = unDynamicBigMap $ sExtra fsStorage
    newExtra = BigMap $ M.insert key (lPackValueRaw v) oldExtra
    newStorage = fsStorage { sExtra = DynamicRecBigMap newExtra }
```